src/Uno.UI.RemoteControl.Host/Extensibility/AddInsExtensions.cs

```diff
@@ -1,17 +1,27 @@
 ﻿using System;
 using System.Linq;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
 using Uno.Utils.DependencyInjection;
 using Uno.UI.RemoteControl.Helpers;
+using Uno.UI.RemoteControl.Server.Telemetry;
 
 namespace Uno.UI.RemoteControl.Host.Extensibility;
 
 public static class AddInsExtensions
 {
 	public static IWebHostBuilder ConfigureAddIns(this IWebHostBuilder builder, string solutionFile)
 	{
-		AssemblyHelper.Load(AddIns.Discover(solutionFile), throwIfLoadFailed: false);
+		return builder.ConfigureServices(services =>
+		{
+			// Get telemetry service for add-in discovery and loading
+			var serviceProvider = services.BuildServiceProvider();
```

Comment:
Is this a recommended pattern to use MS DI? Sounds weird to me to resolve a service during the registration phase.